x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/30_semantic_text_inference.yml

```diff
@@ -0,0 +1,551 @@
+    - match:  { _source.non_inference_field: "non inference test" }
+
+---
+"Reindex works for semantic_text fields":
+  - do:
+      index:
+        index: test-sparse-index
+        id: doc_1
+        body:
+          inference_field: "inference test"
+          another_inference_field: "another inference test"
+          non_inference_field: "non inference test"
+
+  - do:
+      get:
+        index: test-sparse-index
+        id: doc_1
+
+  - set: { _source.inference_field.inference.chunks.0.embeddings: inference_field_embedding }
+  - set: { _source.another_inference_field.inference.chunks.0.embeddings: another_inference_field_embedding }
+
+  - do:
+      indices.refresh: { }
+
+  - do:
+      indices.create:
+        index: destination-index
+        body:
+          mappings:
+            properties:
+              inference_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
+              another_inference_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
+              non_inference_field:
+                type: text
+
+  - do:
+      reindex:
+        wait_for_completion: true
+        body:
+          source:
+            index: test-sparse-index
+          dest:
+            index: destination-index
+  - do:
+      get:
+        index: destination-index
+        id: doc_1
+
+  - match:  { _source.inference_field.text: "inference test" }
+  - match:  { _source.inference_field.inference.chunks.0.text: "inference test" }
+  - match:  { _source.inference_field.inference.chunks.0.embeddings: $inference_field_embedding }
+  - match:  { _source.another_inference_field.text: "another inference test" }
+  - match: { _source.another_inference_field.inference.chunks.0.text: "another inference test" }
+  - match: { _source.another_inference_field.inference.chunks.0.embeddings: $another_inference_field_embedding }
+  - match:  { _source.non_inference_field: "non inference test" }
+
+---
+"Fails for non-existent inference":
+  - do:
+      indices.create:
+        index: incorrect-test-sparse-index
+        body:
+          mappings:
+            properties:
+              inference_field:
+                type: semantic_text
+                inference_id: non-existing-inference-id
+              non_inference_field:
+                type: text
+
+  - do:
+      catch: missing
+      index:
+        index: incorrect-test-sparse-index
+        id: doc_1
+        body:
+          inference_field: "inference test"
+          non_inference_field: "non inference test"
+
+  - match: { error.reason: "Inference id [non-existing-inference-id] not found for field [inference_field]" }
+
+  # Succeeds when semantic_text field is not used
+  - do:
+      index:
+        index: incorrect-test-sparse-index
+        id: doc_1
+        body:
+          non_inference_field: "non inference test"
+
+---
+"Updates with script are not allowed":
```

Comment:
Could you add a test for bulk partial doc updates, upserts?

Also, could you add a test for the `update API`: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html

And tests for update by query API? One where we fail due to scripts I suppose, but another where somebody added the semantic text field AFTER the document was indexed and they want to run an update by query to capture the change? I am not sure this is supported or not, but we need to test it:

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html#picking-up-a-new-property

Reply:
Good idea. I've addressed these tests, please take a look 👍